documentation/src/main/asciidoc/topics/contributing/terminology.adoc

```diff
@@ -0,0 +1,91 @@
+[[terminology]]
+= {brandname} Terminology
+
+{brandname} terminology refers to words and phrases specific to {brandname}
+capabilities and features. It is important to use consistent terminology across
+documentation and the code base to avoid unclear or contradictory language.
+
+Add::
+Add an item to an existing list, group, view, or other container element. For example, "add entries to caches".
+
+Cache::
+A data structure for storing key/value pairs in memory.
+
+Cache configuration::
+A resource that instantiates and controls a cache.
+
+Cache template::
+Generally means a resource from which you create cache configurations. +
+{brandname} provides some pre-defined templates (`org.infinispan.*`). +
+Users can define new templates in `infinispan.xml` with the `*-cache-configuration*` tags. +
+In some APIs, you can use the configuration of an existing cache as a template to configure a new cache.
+
+Cache definition::
+Refers to a named cache instance. To avoid ambiguity, do not use in UI or documentation to mean "cache configuration".
+
+Cache Manager::
+Interface for creating and managing caches. Always spell as two words with capital letters when referring to the abstract notion of a Cache Manager. +
+When referring to specific interfaces, use `CacheManager`, `EmbeddedCacheManager`, or `RemoteCacheManager`.
+
+Clear::
+Delete all the elements of a container. For example, "clear entries in a cache".
+
+Create::
+Build something new. For example, "create a cache".
+
+Delete::
+Remove a single element or a group of similar elements from a container. For example, "delete an entry from the cache" or "delete all expired entries from the cache".
+
+Destroy::
+Do not use to mean "delete".
+
+Index::
+Create an index to improve query performance or add an entry to an existing index when it is added to a cache.
+
+Indexes::
+Plural of index. Do not use "indices".
+
+Off-heap::
+Native memory space outside the managed JVM heap. Always hyphenate when used as an adjective.
+
+Purge::
+Can have negative connotations and is not a preferred term. Use only as a synonym for "clear" or "delete" in subsequent descriptions, such as in the body text of a dialog or text that follows a heading. Do not use as a primary verb or in headings.
+
+Query::
+Look up information within the data set. The words "query" and "search" are synonymous but frequently interchanging the two can lead to inconsistency.
++
+Use in verb form: "Query values in the cache." +
+Use in noun form preceded by Ickle: "Perform an Ickle query on the cache."
+
+Reindex::
+Rebuild an index to update or change an existing index. Do not hyphenate (always "reindex" never "re-index").
+
+Remove::
+Use as a synonym for "clear" or "delete" in subsequent descriptions, such as in the body text of a dialog or text that follows a heading. Do not use as a primary verb or in headings.
```

Comment:
This contradicts the PatternFly terminology list since "delete" and "remove" mean different things: 

Remove: (v.) Use to describe removing an item from a list, group, view, or other container element without completely deleting it. Also see Add and Delete.

Opposite: Add
Notes: If what you’re removing is not readily apparent from the context, consider adding a noun (like “Remove file”). 

We may not want to encourage synonyms since the inconsistency has potential to confuse users (more info in a later comment).

Reply:
Thanks Abi. FYI, I added this based on some notes from our doc so we might want to revisit that before we submit any PRs for the console.

"Remove: a synonym for “clear” or “delete” in the body copy/dialog"

There are also some places, particularly in the REST API, where "remove" is used synonymously with "delete". TBH I'm not sure if the distinction between "remove" and "delete" that you're making will apply to Infinispan.

Generally, when we say "remove" something it's deleted, rather than take it out of a container but it still exists. We even use "remove" in the definition for "delete" in this terminology list.

What if we advise using "permanently remove" in places where there is a need to describe "delete"?

Reply:
Ah, got it. It was one of the verbs you, Alana, and Ingrid defined (that was the day I had to drop the call early). If you feel like it makes sense as a synonym and it suits Infinispan best, then let's go with it 👍